iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -139,36 +142,36 @@ private async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri,
                         // Configure proxy server
                         websocket.Options.Proxy = _amqpTransportSettings.Proxy;
                         if (Logging.IsEnabled)
-                            Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Set ClientWebSocket.Options.Proxy to {_amqpTransportSettings.Proxy}");
+                            Logging.Info(this, $"{nameof(CreateClientWebSocket)} Set ClientWebSocket.Options.Proxy to {_amqpTransportSettings.Proxy}");
                     }
                     catch (PlatformNotSupportedException)
                     {
+                        websocket.Dispose();
                         // Some .NET runtimes don't support this property.
                         if (Logging.IsEnabled)
-                            Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as this framework doesn't support proxy.");
+                            Logging.Error(this, $"{nameof(CreateClientWebSocket)} PlatformNotSupportedException thrown as this framework doesn't support proxy.");
+                        throw new PlatformNotSupportedException("The current .NET runtime does not support setting the proxy.");
```

Comment:
This was closer before.

```suggestion
                        throw new InvalidOperationException("The current .NET runtime does not support setting the proxy.", ex);
```